src/dbnode/client/session.go

```diff
@@ -777,18 +775,12 @@ func (s *session) DedicatedConnection(
 			return
 		}
 
-		newConnFn := s.opts.NewConnectionFn()
-		channel, client, err = newConnFn(channelName, host.Address(), s.opts)
+		channel, client, _, err = establishNewConnection(host.Address(), opts.BootstrappedNodesOnly, s.opts)
 		if err != nil {
 			multiErr = multiErr.Add(err)
 			return
 		}
 
-		if err := s.healthCheckNewConnFn(client, s.opts, opts.BootstrappedNodesOnly); err != nil {
-			multiErr = multiErr.Add(err)
```

Comment:
here we were missing a channel close statement. Now this is fixed by reusing `establishNewConnection`